cmd/credentialUtil.go

```diff
@@ -593,8 +593,24 @@ func createClientOptions(logger common.ILoggerResetable, srcCred *common.ScopedT
 		RetryDelay:    ste.UploadRetryDelay,
 		MaxRetryDelay: ste.UploadMaxRetryDelay,
 	}, policy.TelemetryOptions{
-		ApplicationID: common.AddUserAgentPrefix(common.UserAgent),
+		ApplicationID: buildUserAgentWithTelemetry(common.UserAgent),
 	}, ste.NewAzcopyHTTPClient(frontEndMaxIdleConnectionsPerHost), logOptions, srcCred, reauthCred)
 }
 
 const frontEndMaxIdleConnectionsPerHost = http.DefaultMaxIdleConnsPerHost
+
+// buildUserAgentWithTelemetry creates the user agent string including both
+// environment variable prefix and CLI telemetry parameter
+func buildUserAgentWithTelemetry(baseUserAgent string) string {
+	userAgent := baseUserAgent
+	
+	// Add CLI telemetry parameter if provided
+	if len(azcopyTelemetryValue) > 0 {
+		userAgent = azcopyTelemetryValue + " " + userAgent
+	}
+	
+	// Add environment variable prefix if set
+	userAgent = common.AddUserAgentPrefix(userAgent)
```

Comment:
This existing method was used to take a prefix from env variable and then add userAgent as a string after that. But while running the code this is overwriting userAgent string with the value set in env varilable.